lib/Cake/View/Helper/HtmlHelper.php

```diff
@@ -829,6 +833,19 @@ public function image($path, $options = array()) {
 			unset($options['url']);
 		}
 
+		if (!empty($options['base64']) && !$url) {
+			$fullPath = WWW_ROOT . $path;
+			$file = new File($fullPath, false);
+			if (!$file->exists()) {
+				throw new InvalidArgumentException(__d('cake', 'Unable to find the requested image to output as base64!'));
+			}
+			$file->open();
```

Comment:
`File::open()`, however, will try create the file if it doesn't exist, contrary to what we want. I will add the `File::readable()` check to it though.